Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -203,26 +213,44 @@ private void deriveTaskInfo(
     }
 
     private void deriveTimetable(
-        final ProcessSimulationInfo.ProcessSimulationInfoBuilder builder) {
-
-        builder.timetables(
-            Arrays.asList(Timetable.builder()
-                .defaultTimetable(true)
-                .id(config.getDefaultTimetable().get(CONFIG_DEFAULT_ID_KEY))
-                .name(config.getDefaultTimetable().get(CONFIG_DEFAULT_NAME_KEY))
-                .rules(Arrays.asList(Rule.builder()
-                    .id(UUID.randomUUID().toString())
-                    .name(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_NAME_KEY))
-                    .fromWeekDay(DayOfWeek.valueOf(
-                        config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_FROM_WEEKDAY_KEY)
-                            .toUpperCase(DOCUMENT_LOCALE)))
-                    .toWeekDay(DayOfWeek.valueOf(
-                        config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_TO_WEEKDAY_KEY)
-                            .toUpperCase(DOCUMENT_LOCALE)))
-                    .fromTime(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_FROM_TIME))
-                    .toTime(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_TO_TIME))
-                    .build()))
-                .build()));
+        final ProcessSimulationInfo.ProcessSimulationInfoBuilder builder,
+        final CalendarModel calendarModel) {
+
+        if (calendarModel.getName().equals("Generic 24/7")) {
+            builder.timetables(List.of(createTimetable(
```

Comment:
Hard-coded "Generic 24/7"?

Reply:
We can support this need by using AbsoluteCalendarModel, this is a 24x7 calendar type, adding isAllDayAllTime method to CalendarModel?

Reply:
If no custom calendar is set, PD seems to use the calendar created via `CustomCalendarService.getGenericCalendar()` which in turn uses the `CalendarModel` class and not the `AbsoluteCalendarModel` so if we were to go down this route, we'd need to make some changes to the calendar model package, which I think is better achieved via its own task.
Let me know if my understanding here is incorrect.

Reply:
moved the hardcoded name to a constant.

Reply:
Ok, it seems a little more work to do with CustomCalendar, we can do that later.